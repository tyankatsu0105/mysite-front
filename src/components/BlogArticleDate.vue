<template>
	<div class="BlogArticleDate">
		<font-awesome-icon
			:icon="{ prefix: 'fa', iconName: 'clock' }"
		/>
		<time
			:datetime="time"
			class="BlogArticleDate-DateText"
		>
			{{ text }}
		</time>
	</div>
</template>

<script>
export default {
  props: {
    text: {
      type: String,
      requirer: true,
      default: ""
    },
    time: {
      type: String,
      requirer: false,
      default: ""
    }
  }
};
</script>

<style scoped lang='scss'>
.BlogArticleDate {
  &-DateText {
    margin-left: 5px;
  }
}
</style>
