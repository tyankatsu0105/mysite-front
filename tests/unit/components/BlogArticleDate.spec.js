import { shallowMount } from "@vue/test-utils";
import BlogArticleDate from "@/components/BlogArticleDate.vue";

const propsData = {
  time: `2018.12.15`,
  text: `2018.12.15`
};

describe("BlogArticleDate", () => {
  it("render", () => {
    const wrapper = shallowMount(BlogArticleDate, {
      propsData,
      stubs: ["font-awesome-icon"]
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
