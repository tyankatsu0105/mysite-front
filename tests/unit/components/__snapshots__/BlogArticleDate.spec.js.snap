// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogArticleDate render 1`] = `
<div class="BlogArticleDate">
  <font-awesome-icon-stub icon="[object Object]"></font-awesome-icon-stub> <time datetime="2018.12.15" class="BlogArticleDate-DateText">
    2018.12.15
  </time>
</div>
`;
